### PR TITLE
first draft of in-source doc on CC api

### PR DESCRIPTION
**Pull request by DrXyzzy**
Added to message.coffee:
- general comments about API messages
- specific docs and examples for `ping`, `create project`, and `get_usernames`.

### CODE REVIEW

File: src/smc-util/message.coffee [hunk 1]
```diff
@@ -195,10 +217,21 @@ message
 #############################################
 
 API message
-    event       : 'get_usernames'
+    event       : 'get_usernames' # get first and last names for a list of account ids
     id          : undefined
     account_ids : required
 
+    # Note: Options for this messsge must be sent as JSON object.
```
Comment (williamstein) [inline src/smc-util/message.coffee]: typo: ' messsge' -->  message

File: src/smc-util/message.coffee [hunk 2]
```diff
@@ -195,10 +217,21 @@ message
...
+    # example (reformatted for readability):
```
Comment (williamstein) [inline src/smc-util/message.coffee]: Use ### for multiline coffeescript comments, e.g.
###
a bunch
of lines
###

File: src/smc-util/message.coffee [hunk 3]
```diff
@@ -448,7 +481,7 @@ message
 # starting_bundle_number.
 #
 # client --> hub --> project_server
-API message
+API message # unimplemented
```
Comment (williamstein) [inline src/smc-util/message.coffee]: Why do you say "unimplemented?"  It is too implemented.

File: src/smc-util/message.coffee [hunk 4]
```diff
@@ -775,10 +813,24 @@ message
 ###
 Ping/pong -- used for clock sync, etc.
 ###
+
 API message
-    event : 'ping'
+    event : 'ping' # test connection to API back end
```
Comment (williamstein) [inline src/smc-util/message.coffee]: Returns time as ISO string when server responds to ping

File: src/smc-util/message.coffee [hunk 5]
```diff
@@ -798,12 +850,12 @@ logged in.
 # return a JSON object with all data that is
 # meant to be publically available about this project,
 # who owns it, the title/description, etc.
-API message
+API message # unimplemented
```
Comment (williamstein) [inline src/smc-util/message.coffee]: why  unimplemented ?

If anything is genuinely unimplemented, then it should just be deleted.

File: src/smc-util/message.coffee (hunk 3 above)
Comment (DrXyzzy) [inline src/smc-util/message.coffee]: > Why do you say "unimplemented?" It is too implemented.

I got unknown message event errors and didn't see the message event type elsewhere in the source using `git grep`.
```
 {"error":"unknown message event type 'public_get_project_info'"}
 {"error":"unknown message event type 'public_project_info'"}
 {"error":"unknown message event type 'save_project'"}
```

Comment (williamstein) [inline src/smc-util/message.coffee]: OK, they must have got removed from the implementation; I think they should get deleted entirely then, and that functionality is now provided in a different way (i.e., via the query message type).

**Comment by DrXyzzy** [comment]
oops, forgot this: Ref: #1987

**Comment by williamstein** [comment]
What's the status of this?  I could just merge it; my only fear is that we were wrong about deleted messages (but I don't think we are).  Are you going to rewrite it to use message2 today, or later...?

**Comment by DrXyzzy** [comment]
See new pr #2018 which uses API message2 and includes recent changes from master. This PR shouldn't be merged.

**Pull request closed without merging**